services/wallet/reader.go

```diff
@@ -250,6 +250,26 @@ func (r *Reader) GetWalletToken(ctx context.Context, addresses []common.Address)
 		availableNetworks = append(availableNetworks, network)
 	}
 
+	cachedTokens, err := r.GetCachedWalletTokensWithoutMarketData()
+	if err != nil {
+		return nil, err
+	}
+
+	isCachedToken := func(address common.Address, symbol string, chainID uint64) bool {
```

Comment:
`GetWalletToken` is chunky enough already, perhaps move this definition to a separate named function?